sycl/test-e2e/ESIMD/raw_send.cpp

```diff
@@ -0,0 +1,181 @@
+  src0_ref2 = data.template bit_cast_view<unsigned int>();
+  uint32_t exDesc = 0xA;
+  SurfaceIndex desc = esimd::get_surface_index(acc);
+  desc += 0x40A0200;
+  constexpr uint8_t execSize = 0x83;
+  constexpr uint8_t sfid = 0x0;
+  constexpr uint8_t numSrc0 = 0x2;
+#ifdef NEW_API
+  return experimental::esimd::raw_send(src0, exDesc, desc, execSize, sfid,
+                                       numSrc0);
+#else
+  return experimental::esimd::raw_send_store(src0, exDesc, desc, execSize, sfid,
+                                             numSrc0);
+#endif
+}
+
+template <typename T> int test(queue q) {
+  constexpr unsigned Size = 1024 * 128;
+  constexpr unsigned VL = sizeof(T) == 4 ? 16 : 32;
+  T *A = new T[Size];
+  T *B = new T[Size];
+  T *C = new T[Size];
+
+  for (unsigned i = 0; i < Size; ++i) {
+    A[i] = B[i] = i;
+    C[i] = 0;
+  }
+
+  try {
+    buffer<T, 1> bufa(A, range<1>(Size));
+    buffer<T, 1> bufb(B, range<1>(Size));
+    buffer<T, 1> bufc(C, range<1>(Size));
+
+    // We need that many workgroups
+    range<1> GlobalRange{Size / VL};
+
+    // We need that many threads in each group
+    range<1> LocalRange{1};
+
+    auto e = q.submit([&](handler &cgh) {
+      auto PA = bufa.template get_access<access::mode::read>(cgh);
+      auto PB = bufb.template get_access<access::mode::read>(cgh);
+      auto PC = bufc.template get_access<access::mode::write>(cgh);
+      cgh.parallel_for(
+          GlobalRange * LocalRange, [=](id<1> i) SYCL_ESIMD_KERNEL {
+            unsigned int offset = i * VL * sizeof(T);
+            simd<T, VL> va = dwaligned_block_read<T, VL>(PA, offset);
+            simd<T, VL> vb = dwaligned_block_read<T, VL>(PB, offset);
+            simd<T, VL> vc = va + vb;
+            constexpr int SIZE = VL / 2;
+            block_write1(PC, offset, vc.template select<SIZE, 1>(0).read());
+            offset += SIZE * sizeof(T);
+            block_write2(PC, offset, vc.template select<SIZE, 1>(SIZE).read());
+          });
+    });
+    e.wait();
+  } catch (sycl::exception const &e) {
+    std::cout << "SYCL exception caught: " << e.what() << '\n';
+
+    delete[] A;
+    delete[] B;
+    delete[] C;
+    return 1;
+  }
+
+  int err_cnt = 0;
+
+  for (unsigned i = 0; i < Size; ++i) {
+    if (A[i] + B[i] != C[i]) {
+      if (++err_cnt < 10) {
+        std::cout << "failed at index " << i << ", " << C[i] << " != " << A[i]
+                  << " + " << B[i] << "\n";
+      }
+    }
+  }
+
+  delete[] A;
+  delete[] B;
+  delete[] C;
+
+  std::cout << (err_cnt > 0 ? "FAILED\n" : "Passed\n");
+  return err_cnt;
+}
+
+int main(void) {
+
+  queue q(esimd_test::ESIMDSelector, esimd_test::createExceptionHandler());
+
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<sycl::info::device::name>()
+            << "\n";
+  int err_cnt = 0;
+
+  err_cnt += test<sycl::ext::intel::experimental::esimd::tfloat32>(q);
+  err_cnt += test<sycl::half>(q);
```

Comment:
For half types we may need a check:   if (dev.has(sycl::aspect::fp16)) before running the check.